cpp/examples/tutorial_examples/arrow_example.cc

```diff
@@ -0,0 +1,94 @@
+#include <arrow/api.h>
+#include <arrow/result.h>
+#include <arrow/status.h>
+#include <arrow/table.h>
+
+#include <iostream>
+
+using arrow::Status;
+
+namespace { 
+
+Status RunMain(int argc, char** argv) {
+
+  // Creating Arrays and Tables
```

Comment:
Probably add the comment for each block of code corresponding to a structure type being created.

Reply:
I added comments to guide through, though I may need to revert some of them once there's some supporting prose.